rust/azure_iot_operations_connector/src/base_connector/managed_azure_device_registry.rs

```diff
@@ -380,31 +331,147 @@ impl DeviceEndpointClient {
     }
 
-    /// Used to receive updates for the Device/Inbound Endpoint from the Azure Device Registry Service.
-    /// This function returning `Some(())` indicates that the device specification has been
-    /// updated in place. The function returns [`None`] if there will be no more notifications.
+    /// Used to receive notifications related to the Device/Inbound Endpoint
+    /// from the Azure Device Registry Service.
     ///
-    /// TODO: add deletion monitoring to this same receive flow
+    /// Returns [`ClientNotification::Updated`] if the Device Endpoint
+    /// Specification has been updated in place.
+    ///
+    /// Returns [`ClientNotification::Deleted`] if the Device Endpoint has been
+    /// deleted. The [`DeviceEndpointClient`] should not be used after this
+    /// point, and no more notifications will be received.
+    ///
+    /// Returns [`ClientNotification::Created`] with a new [`AssetClient`] if a new
+    /// Asset has been created.
     ///
     /// # Panics
     /// If the Azure Device Registry Service provides a notification that isn't for this Device Endpoint. This should not be possible.
     ///
     /// If the specification mutex has been poisoned, which should not be possible
-    pub async fn recv_update(&mut self) -> Option<()> {
-        // handle the notification
-        // We set auto ack to true, so there's never an ack here to deal with. If we restart, then we'll implicitly
-        // get the update again because we'll pull the latest definition on the restart, so we don't need to get
-        // the notification again.
-        let (updated_device, _) = self.device_update_observation.recv_notification().await?;
-        // update self with updated specification
-        let mut unlocked_specification = self.specification.write().unwrap(); // unwrap can't fail unless lock is poisoned
-        *unlocked_specification = DeviceSpecification::new(
-                updated_device,
-                // TODO: get device_endpoint_credentials_mount_path from connector config
-                "/etc/akri/secrets/device_endpoint_auth",
-                &self.device_endpoint_ref.inbound_endpoint_name,
-            ).expect("Device Update Notification should never provide a device that doesn't have the inbound endpoint");
-        Some(())
+    pub async fn recv_notification(&mut self) -> ClientNotification<AssetClient> {
+        loop {
+            tokio::select! {
+                biased;
+                update = self.device_update_observation.recv_notification() => {
+                    // handle the notification
+                    // We set auto ack to true, so there's never an ack here to deal with. If we restart, then we'll implicitly
+                    // get the update again because we'll pull the latest definition on the restart, so we don't need to get
+                    // the notification again.
+                    let Some((updated_device, _)) = update else {
```

Comment:
Is this `let Some...` because of clippy? Would a match make more sense on update?